src/FSharpLint.Core/Rules/Conventions/RecursiveAsyncFunction.fs

```diff
@@ -31,6 +31,23 @@ let private getFunctionNameFromAsyncCompExprBinding = function
 
 let checkRecursiveAsyncFunction (args:AstNodeRuleParams) (range:Range) (doBangExpr:SynExpr) breadcrumbs =
     let doTokenRange = Range.mkRange "do!" (Position.mkPos range.StartLine range.StartColumn) (Position.mkPos range.StartLine (range.StartColumn + 3))
+
+    let suggestFix () = 
+        let suggestedFix = lazy(
+            ExpressionUtilities.tryFindTextOfRange doTokenRange args.FileContent
+            |> Option.map (fun fromText ->
+                { FromText = fromText
+                  FromRange = doTokenRange
+                  ToText = "return!" }))
+
+        {
+            Range = range
+            Message = Resources.GetString("RulesConventionsRecursiveAsyncFunctionError")
+            SuggestedFix = Some suggestedFix
+            TypeChecks = []
```

Comment:
@Mersho this seems to be a bug in the FavourStaticEmptyFields false negative rule, let's create a PR for it